spinoso-string/src/lib.rs

```diff
@@ -453,25 +301,8 @@ impl String {
     /// ```
     #[inline]
     #[must_use]
-    pub const fn encoding(&self) -> Encoding {
-        self.encoding
-    }
-
-    /// Set the [`Encoding`] of this `String`.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use spinoso_string::{Encoding, String};
-    ///
-    /// let mut s = String::utf8(b"xyz".to_vec());
-    /// assert_eq!(s.encoding(), Encoding::Utf8);
-    /// s.set_encoding(Encoding::Binary);
-    /// assert_eq!(s.encoding(), Encoding::Binary);
-    /// ```
-    #[inline]
-    pub fn set_encoding(&mut self, encoding: Encoding) {
-        self.encoding = encoding;
+    pub fn encoding(&self) -> Encoding {
+        self.inner.encoding()
```

Comment:
I mentioned elsewhere, but I removed the `set_encoding` since it didn't feel like forcing one EncodedString struct into the other made sense - however I think I should re-implement due to at least `String#force_encoding`. Thoughts?

Reply:
Yup `set_encoding` will have to be reimplemented to `String#force_encoding`, but we can do that after landing this. I'll file a ticket.

Reply:
Filed:

- https://github.com/artichoke/artichoke/issues/1719